presto-password-authenticators/src/main/java/io/prestosql/plugin/password/salesforce/SalesforceBasicAuthenticator.java

```diff
@@ -0,0 +1,187 @@
+ * the Salesforce security token at https://help.salesforce.com/articleView?id=user_security_token.htm.
+ * <br><br>
+ * If your password is <code>foo</code> and token is <code>bar</code>, then your Presto password will be <code>foobar</code>.
+ * <br><br>
+ * Admins can configure one or more Salesforce Organization Ids (18 char) via <code>salesforce.org</code> in <code>password-authenticator.properties</code>
+ * to act as a single layer of authZ.  Essentially, if the user who logs in is not part of the configured org, their access will be denied.  Alternatively,
+ * the admin can specify "all", rather than actual orgs.  This will allow any authenticated Salesforce user access to Presto.
+ */
+public class SalesforceBasicAuthenticator
+        implements PasswordAuthenticator
+{
+    private static final Logger log = Logger.get(SalesforceBasicAuthenticator.class);
+
+    private Set<String> orgs;                                   // Set of Salesforce orgs, which users must belong to in order to authN.
+    private Map<String, String> responseMap = new HashMap<>();  // A flattened map of xml elements from the Salesforce login response.
+    private String key = "";                                    // Will hold the xml element tag, we can add to the map with its text attribute.
+    private final HttpClient httpClient;                        // An http client for posting to the Salesforce login endpoint.
+    private final LoadingCache<Credential, Principal> userCache;
+
+    private final Locale locale = Locale.US;                    // Tested API request and response for user with Japanese locale and language preference,
+    // and responses are English, and organization id is not in Japaneses characters (this is
+    // also true of the organization id in the UI, even with other text showing in Japanese).
+
+    @Inject
+    public SalesforceBasicAuthenticator(SalesforceConfig config, @SalesforceAuthenticationClient HttpClient httpClient)
+    {
+        this.orgs = config.getOrgSet();
+        this.httpClient = httpClient;
+
+        this.userCache = CacheBuilder.newBuilder()
+                .maximumSize(config.getCacheSize())
+                .expireAfterWrite(config.getCacheExpireSeconds(), TimeUnit.SECONDS)
+                .build(CacheLoader.from(this::doLogin));
+    }
+
+    @Override
+    public Principal createAuthenticatedPrincipal(String username, String password)
+    {
+        try (ThreadContextClassLoader ignored = new ThreadContextClassLoader(getClass().getClassLoader())) {
+            return userCache.getUnchecked(new Credential(username, password));
+        }
+        catch (UncheckedExecutionException e) {
+            throwIfInstanceOf(e.getCause(), AccessDeniedException.class);
+            throw e;
+        }
+    }
+
+    // This does the work of logging into Salesforce.
+    private Principal doLogin(Credential credential)
+    {
+        log.info("Logging into Salesforce.");
```

Comment:
Had it there for a couple of reasons.  First is simply to ensure the cache was working as expected, so you're right, definitely not a production need.  The second is that at Salesforce we are using multiple authentication options, so that users can login either via credentials we store in a password file, or through another custom mechanism we wrote, or via their Salesforce credentials.  This is a little bit of a crude audit mechanism, though clearly very crude.

Happy to remove it, or is it acceptable to change to `debug`?